public/app/features/alerting/unified/components/contact-points/ContactPoints.v2.tsx

```diff
@@ -225,6 +237,36 @@ const ContactPointsList = ({
   );
 };
 
+// let's search in two different haystacks, the name of the contact point and the type of the receiver(s)
+function useContactPointsSearch(
+  contactPoints: ContactPointWithMetadata[],
+  search?: string
+): ContactPointWithMetadata[] {
+  const finder = useMemo(() => new uFuzzy({ unicode: true }), []);
```

Comment:
why do we need useMemo in here? I think we can create this finder const outside the hook without using any memo
`const finder = new uFuzzy({ unicode: true });`
or there is a reason for it?

Reply:
and is there a reason for not using in uFuzzy options?
```
 {intraIns: 1,
  intraSub: 1,
  intraTrn: 1,
  intraDel: 1,}
```

Reply:
I don't actually know what it initializes.

If there are mutations of internal data structures inside the class instance (assumption) I'd rather not create a global instance where multiple invocations of the hook are overwriting values. Granted, since JS is single threaded this might not be a real problem but I'd like to avoid globally scoped variables wherever I can.

What do those fuzzy options do? I just went with the defaults but if those are good options to tweak I don't see why not :D

Reply:
nice!